.github/workflows/ci-test.yml

```diff
@@ -277,6 +277,11 @@ jobs:
         with:
           ref: ${{ github.event.pull_request.head.sha }}
       - uses: ./.github/actions/build-setup
+      - uses: actions/cache@v2.1.4
+        with:
+          path: "/opt/cargo/git\n/opt/cargo/registry\n/opt/cargo/.package-cache"
+          key: crates-${{ runner.os }}-${{ hashFiles('Cargo.lock') }}
+          restore-keys: "crates-${{ runner.os }}"
```

Comment:
why there are difference between key (which based off hash of Cargo.lock) and restore-keys (doesn't) for the actions cache?

Reply:
oh, nvm.   Found it here https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows

```
An ordered list of alternative keys to use for finding the cache if no cache hit occurred for key.
```